pac4j-jwt/src/main/java/org/pac4j/jwt/credentials/authenticator/JwtAuthenticator.java

```diff
@@ -202,11 +199,19 @@ protected void createJwtProfile(final TokenCredentials credentials, final JWT jw
         if (subject == null) {
             throw new TechnicalException("JWT must contain a subject ('sub' claim)");
         }
-
         if (!subject.contains(CommonProfile.SEPARATOR)) {
             subject = JwtProfile.class.getName() + CommonProfile.SEPARATOR + subject;
         }
 
+        final Date expirationTime = claimSet.getExpirationTime();
+        if (expirationTime != null) {
+            final Date now = new Date();
+            if (expirationTime.before(now)) {
+                logger.error("The JWT is expired: no profile is built");
+                return;
```

Comment:
Good question, I hesitated on this.

If you throw a `TechnicalException`, the login flow will be broken and a 500 error page will be returned. Otherwise, returning `null` or throwing a `CredentialsException` will return a 401 error page.

I tried to organize the error cases in two main categories:

- if it's an expected issue (valid JWT), I want to return a 401 error page: no appropriate encryption / signature configuration, expired token

- if it's an unexpected issue, I want to return a 500 error page: no `sub` claim, not a valid token.

Though, I don't have a strong feeling on this.